src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp

```diff
@@ -182,114 +273,162 @@ class npc_announcer_toc5 : public CreatureScript
 
             if (uiPointId == 1)
                 me->SetFacingTo(ORIENTATION);
+            else if (uiPointId == 2)
+                NextStep(500, false, 20);
         }
 
-        void DoSummonGrandChampion(uint32 uiBoss)
+        void DoSummonNextGrandChampion(bool skipEvent = false)
```

Comment:
look what i've done in https://github.com/Aokromes/TrinityCore/commit/5696cef3f499dd3aceeecdeb5fc9741a06853676 it is a bit better